script.js

```diff
@@ -168,137 +254,121 @@ function addTemp() {
+        if (x > 0 && y > 0 && y < canvasHeight && x < canvasWidth) {
+            return true;
+        } else {
+            return false;
+        }
     }
-    // Update canvas if needed
-    updateDrawing();
-}
 
-// Draws AT, with gradient if AT is enabled
-function drawAT(midpoint, y) {
-    // If AT not active then no gradient
-    let colorAT = "#55066d";
-    // change later to AT 
-    if (activeAT === true) {
-        colorAT = gradientAT;
+class Molecule {
+
+    constructor(moleculeType, startX, startY) {
+        this.x = startX;
+        this.y = startY;
     }
 
-    // Draw arcs for AT
-    drawArc(midpoint-45, y-25, midpoint, y+50, midpoint+60, y-30, colorAT);
-    drawArc(midpoint-45, y+25, midpoint, y-50, midpoint+60, y+30, colorAT);
-    drawArc(midpoint-50, y+15, midpoint+10, y-60, midpoint+65, y-10, colorAT);
-    drawArc(midpoint-50, y-15, midpoint+10, y+60, midpoint+65, y+10, colorAT);
-}
+    /************************************/
 
-// Draws a closed port
-function drawPort(x, y) {
-    drawCircle((wallWidth/2) + x, 10 + y, wallWidth/2, "orange");
-    drawRectangle(x, 10 + y, wallWidth, 75, "orange");
-    drawCircle((wallWidth/2) + x, 85 + y, wallWidth/2, "orange");
-}
+    /************************************/
 
-// Draws walls depending on what is enabled/disabled
-function drawWall() {
-    // Dark walls
-    drawRectangle(wallX, 0, wallWidth, 125, "grey");
-    drawRectangle(wallX, 220, wallWidth, 60, "grey");
-    drawRectangle(wallX, 375, wallWidth, 125, "grey");
-    
-    // Draw ports if ports are open
-    if (port1Open === true) {
-        drawPort(wallX, 125);
+    static isCollidingWithWallOrPort() {
+        // add code
+        return false;
     }
-    if (port2Open === true) {
-        drawPort(wallX, 280);
-    }
-    
-    // Draws active transport
-    drawAT(canvasWidth/2, 50);
-    drawAT(canvasWidth/2, 450);
-}
 
-// Returns a negative or positive number for random movement
-function randomCo() {
-    // Speed + random value so there's always some movement
-    let distance = speed + (Math.random() * Math.floor(speed));
-    
-    // Picks a direction of negative or positive
-    let direction = (Math.random());
-    if (direction > 0.5) {
-        direction = 1;
+    static isCollidingWithAT() {
+        // add code
+        return false;
     }
-    else {
-        direction = -1;
+
+    /************************************/
+
+    updateMolecule() {
+        let tempX, tempY;
+            /* Add x y ranges for checks so lil particles far right/left don't have to go
+            through the checks for the wall/AT and particles high up won't have to go 
+            through checks for lower AT*/
+            
+        do {
+                // Picks a new location for the particle
+                tempX = this.x + randomCo();
+                tempY = this.y + randomCo();
+                // If new location is outside canvas, loop
+        } 
+        while (!isInCanvas(tempX, tempY));
```

Comment:
I think this is where we can add the checks for collision, since `isInCanvas()` is basically doing that for the edge of the screen.